src/lib/classes/lsp3-universal-profile.spec.ts

```diff
@@ -1,95 +1,191 @@
-          universalReceiverDelegateInit: baseContracts.UniversalReceiverDelegate?.address,
+      });
+
+      let events;
+
+      deployments$.subscribe({
+        next: (deploymentEvents: DeploymentEvent) => {
+          events = deploymentEvents;
         },
-      }
-    );
+        error: (error) => {
+          expect(1).toEqual(error);
+          done();
+        },
+        complete: async () => {
+          const ownerAddress = await events.ERC725Account.contract.owner();
+          // const keyManagerAddress = await events.KeyManager.contract.address;
+          expect(ownerAddress).toEqual('0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266');
+
+          await events.ERC725Account.contract.setData(
+            '0x5ef83ad9559033e6e941db7d7c495acdce616347d28e90c7ce47cbfcfcad3bc5',
+            '0xbeefbeef',
+            {
+              from: await signer.getAddress(),
+            }
+          );
+          const data = await events.ERC725Account.contract.getData(
+            '0x5ef83ad9559033e6e941db7d7c495acdce616347d28e90c7ce47cbfcfcad3bc5'
+          );
 
-    const ownerAddress = await contracts.ERC725Account.owner();
-    expect(ownerAddress).toEqual('0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266');
+          expect(data).toEqual('0xbeefbeef');
+          done();
+        },
+      });
+    });
+    it('should deploy and set LSP3Profile data', async () => {
+      const ownerAddress = await universalProfile.owner();
+      expect(ownerAddress).toEqual(keyManager.address);
+
+      const data = await universalProfile.getData([
+        '0x5ef83ad9559033e6e941db7d7c495acdce616347d28e90c7ce47cbfcfcad3bc5',
+      ]);
+
+      expect(data[0].startsWith('0x6f357c6a')).toBe(true);
+    });
+  });
+
+  describe('Deploying a UP with one controller address', () => {
+    let uniqueController: SignerWithAddress;
+    let universalProfile;
+
+    beforeAll(async () => {
+      uniqueController = signers[0];
+
+      const { ERC725Account } = await lspFactory.LSP3UniversalProfile.deploy({
+        controllerAddresses: [uniqueController.address],
+      });
+
+      universalProfile = UniversalProfile__factory.connect(ERC725Account.address, uniqueController);
+    });
 
-    const data = await contracts.ERC725Account.getData([
-      '0x5ef83ad9559033e6e941db7d7c495acdce616347d28e90c7ce47cbfcfcad3bc5',
-    ]);
+    it('controller address should have DEFAULT_PERMISSIONS set', async () => {
+      const [signerPermissions] = await universalProfile.getData([
+        PREFIX_PERMISSIONS + uniqueController.address.substring(2),
+      ]);
 
-    expect(data[0].startsWith('0x6f357c6a')).toBe(true);
+      expect(signerPermissions).toEqual(ERC725.encodePermissions(DEFAULT_PERMISSIONS));
+    });
+
+    it('controller address should be registered in AddressPermissions[0] array', async () => {
+      const hexIndex = ethers.utils.hexlify([0]);
+      const leftSide = ADDRESS_PERMISSIONS_ARRAY_KEY.slice(0, 34);
```

Comment:
I think these can be cleaned up a bit by removing the `leftSide` `rightSide` variables and just joining them when you declare the key. 
e.g.
`const permissionKey = ADDRESS_PERMISSIONS_ARRAY_KEY.slice(0, 34) + ethers.utils.hexZeroPad(hexIndex, 16)`